main/views.py

```diff
@@ -213,6 +221,38 @@ def form_valid(self, form):
             return redirect('signup')
 
 
+def space_approval(request, key, action):
+
+    if action != 'approve' and action != 'reject':
+        # this shouldn't happen - just redirect to home
+        return redirect('/')
+
+    # lookup user info based on key
+    users = User.objects.filter(space_request_key=key)
+
+    # should only get one object back
+    if users.count() == 1:
```

Comment:
(There's also a kind of neat trick with this one - you can do `except User.DoesNotExist:` and that will only catch the exceptions where users don't exist.)